lib/rules/no-extra-parens.js

```diff
@@ -256,6 +256,22 @@ module.exports = {
         }
 
         /**
+         * Determines whether a node should be followed by an additional space when removing parens
+         * @param {ASTNode} node node to evaluate; must be surrounded by parentheses
+         * @returns {boolean} `true` if a space should be inserted after the node
+         * @private
+         */
+        function requiresAfterSpace(node) {
+            const nextTwoTokens = sourceCode.getTokensAfter(node, { count: 2 });
+            const rightParenToken = nextTwoTokens[0];
+            const tokenAfterRightParen = nextTwoTokens[1];
+
+            return rightParenToken && tokenAfterRightParen &&
+                !sourceCode.isSpaceBetweenTokens(rightParenToken, tokenAfterRightParen) &&
+                tokenAfterRightParen.type === "Keyword";
```

Comment:
Instead of checking whether the rightmost token is a keyword, can you use something like this instead?

```js
const tokenBeforeRightParen = sourceCode.getLastToken(node);
!astUtils.canTokensBeAdjacent(tokenBeforeRightParen, tokenAfterRightParen)
```

The reason a space is needed isn't because the following token is a keyword, it's because the two tokens will combine if the parens are removed. This makes a difference in cases like this:

```js
for ((foo['bar'])of baz);
```

The `]` and `of` would not combine into a single token if the parens were removed, but the autofixer unnecessarily inserts a space at the moment.